ProcessLib/ComponentTransport/ComponentTransportFEM.h

```diff
@@ -341,35 +344,41 @@ class LocalAssemblerData : public ComponentTransportLocalAssemblerInterface
                            solute_dispersivity_transverse *
                                velocity_magnitude) *
                           I);
+            const double R_times_phi(retardation_factor * porosity);
+            GlobalDimVectorType const mass_density_flow = velocity * density;
+            auto const N_t_N = (N.transpose() * N).eval();
+            if (_process_data.non_advective_form)
+            {
+                const double C_int_pt(N.dot(C_nodal_values));
+                MCp.noalias() += N_t_N * (C_int_pt * R_times_phi * drho_dp * w);
+                MCC.noalias() += N_t_N * (C_int_pt * R_times_phi * drho_dC * w);
+                KCC.noalias() -= dNdx.transpose() * mass_density_flow * N * w;
+            }
+            else
+            {
+                BC.noalias() -=
+                    N.transpose() *
+                    (mass_density_flow.dot(dNdx * C_nodal_values) * w);
+            }
+            MCC.noalias() += N_t_N * (R_times_phi * density * w);
+            KCC.noalias() += dNdx.transpose() * hydrodynamic_dispersion * dNdx *
+                                 (density * w) +
+                             N_t_N * (decay_rate * porosity *
+                                      retardation_factor * density * w);
```

Comment:
You can reuse `R_times_phi` (` = porosity*retardation_factor`) here.

Reply:
changed